pkg/roles/roles.go

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package roles
+
+import (
+	"strings"
+
+	galasaErrors "github.com/galasa-dev/cli/pkg/errors"
+)
+
+func validateRoleName(nameToValidate string) (string, error) {
+	var err error
+	name := strings.TrimSpace(nameToValidate)
+
+	if name == "" || strings.ContainsAny(name, " .\n\t") || !isLatin1(name) {
+		err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_INVALID_ROLE_NAME)
+	}
+	return name, err
+}
+
+// Checks if a given string contains only characters in the Latin-1 character set (codepoints 0-255),
+// returning true if so, and false otherwise
+func isLatin1(str string) bool {
+	isValidLatin1 := true
+	for _, character := range str {
+		if !((character >= 'a' && character <= 'z') ||
+			(character >= 'A' && character <= 'Z') ||
+			(character >= '0' && character <= '9')) {
+			isValidLatin1 = false
+			break
+		}
```

Comment:
Isn't this only checking for alphanumeric characters rather than Latin-1? I thought Latin-1 allows for other special characters along with a-z, A-Z, and 0-9

Reply:
It includes things like % and ' ' (space) which I thought we could do without in a role name, given it will be travelling to the server as a query parameter, and we can avoid having to uuencode it this way.

Yes the function name needs re-work.